logging/data_logger.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Liam Kelly on 11/7/18.
+//
+
+#ifndef NVIDIA_DATA_LOGGER_H
+#define NVIDIA_DATA_LOGGER_H
+
+#include "log_entry.h"
+
+#define ENTRY_BUFFER_SIZE 200 // After we reach this many entries in memory, log them.
```

Comment:
we can play around with this buffer size depending on our logging frequency, but i figured this would be a reasonable size to start with